jwst/cube_build/cube_build_step.py

```diff
@@ -530,3 +543,55 @@ def read_user_input(self):
 # remove duplicates if needed
             self.pars_input['grating'] = list(set(self.pars_input['grating']))
 # ________________________________________________________________________________
+
+    def check_offset_file(self):
+        """Read in an optional ra and dec offsets for each file.
+
+        Summary
+        ----------
+        Check that is file is asdf file.
+        check the file has the correct format:
+        For each file in the input  assocation check that there is a corresponding
+        file in the offset file.
+        Also check that each file in the offset list contain a ra offset and dec offset.
+
+       """
+
+        # validate the offset file using the schema file
+        DATA_PATH = Path(__file__).parent
+        
+        try:
+            af = asdf.open(self.offset_file, custom_schema=DATA_PATH/'ifuoffset.schema.yaml')
+        except:
+            self.log.error('Validation Error for offset file')
+            self.log.error('Turning off adjusting by offsets')
+            return None
+
+
+        offset_filename = af['filename']
+        offset_ra = af['raoffset']
+        offset_dec = af['decoffset']
+        offset_unit = af['units']
+
+        if offset_unit != 'arcsec':
+            self.log.error('Provide the offset units in units of arcsec ')
+            self.log.error('Turning off adjusting by offsets ')
+            af.close()
+            return None
+
+        for model in self.input_models:
+            file_check = model.meta.filename
+            if file_check in offset_filename:
+                continue
```

Comment:
Here, we should probably check for and discard any paths in the filenames before matching them (e.g. with os.path.basename).

Reply:
I have checked that if the association has a filename with a path then the file_check value is just the filename
It is handled in how the association sets model.meta.filename
The issue with the path is if the offset file has the path. I put in the documentation that the offset filename can not have a path in the filename. 
Is this sufficient  or where you thinking of something else ?

Reply:
Good to know that model.meta.filename will only contain the basename, though!  Thanks for checking that, I wasn't sure.